srf_generation/input_file_generation/realisation_to_srf.py

```diff
@@ -1,68 +1,92 @@
 
-import yaml
-from h5py import File as h5open
 import numpy as np
-from os import makedirs, path, remove
-from qcore import binary_version, srf, geo, qclogging, utils
+import yaml
+import h5py
+from qcore import binary_version, geo, qclogging, srf, utils
+from qcore.uncertainties import mag_scaling
+from qcore.uncertainties.mag_scaling import MagnitudeScalingRelations
 from qcore.utils import compare_versions
-from qcore.uncertainties.mag_scaling import (
-    mag2mom,
-    MagnitudeScalingRelations,
-    mw_to_a_skarlatoudis,
-)
-
-from srf_generation.pre_processing_common import (
-    calculate_corners,
-    get_hypocentre,
-    load_realisation_file_as_dict,
-)
+from srf_generation import pre_processing_common
 from srf_generation.source_parameter_generation.common import (
     DEFAULT_1D_VELOCITY_MODEL_PATH,
 )
 from srf_generation.source_parameter_generation.uncertainties.common import (
-    HF_RUN_PARAMS,
     BB_RUN_PARAMS,
+    HF_RUN_PARAMS,
     LF_RUN_PARAMS,
 )
 
-
 SRF_SUBFAULT_SIZE_KM = 0.1
 
 SRF2STOCH = "srf2stoch"
 GENERICSLIP2SRF = "generic_slip2srf"
 FAULTSEG2GSFDIPDIR = "fault_seg2gsf_dipdir"
 
 CORNERS_HEADER = (
-    "> header line here for specifics \n\
-> This is the standard input file format where the \
-hypocenter is first then for each \n\
->Hypocenter (reference??) \n",
-    "> Below are the corners \
-(first point repeated as fifth to close box \n",
+    """> header line here for specifics
+> This is the standard input file format where the hypocenter is first then for each
+> Hypocenter (reference??) """,
+    "> Below are the corners (first point repeated as fifth to close box)\n",
 )
 
 
-def get_n(fault_size, sub_fault_size):
-    return int(round(fault_size / sub_fault_size))
+def number_subdivisions(fault_size: float, subdivision_size: float) -> int:
+    """
+    Calculate the number of meshgrid subdivisions for each fault plane.
+
+    Parameters
+    ----------
+    fault_size : float
+        The size of the fault.
+    subdivision_size : float
+        The size of each subgrid
+
+    Returns
+    -------
+    int
+        The number of subdivisions
+    """
+
+    return round(fault_size / subdivision_size)
 
 
 def create_stoch(
-    stoch_file,
-    srf_file,
-    single_segment=False,
+    stoch_file: str,
+    srf_file: str,
+    single_segment: bool = False,
     logger: Logger = qclogging.get_basic_logger(),
 ):
+    """Create a stoch file from a SRF file.
+
+    Parameters
+    ----------
+    stoch_file : str
+        The filepath to output the stoch file to.
+    srf_file : str
+        The filepath of the SRF file.
+    single_segment : bool
+        True if the stoch file is a single segment.
+    logger : Logger
+        Optional alternative logger for log output.
     """
-    Creates stoch file from srf file.
-    """
+
     logger.debug("Generating stoch file")
-    out_dir = path.dirname(stoch_file)
-    makedirs(out_dir, exist_ok=True)
+    out_dir = os.path.dirname(stoch_file)
```

Comment:
Not an issue, but as a note any new code we like to use pathlib's Path instead of os.path

Reply:
Yeah, I am going to write the type-5 SRF gen code in a new module and follow the new best practices there. Then, after I've done that and worked out how best to write this generation code, I'll come back and rewrite the other cases in a way that makes it possible to read and understand. There is way too much cleanup work for one pull request here.